### PR TITLE
autolev: removed numpy and math imports from parser outputs

### LINKED ISSUE #15186: Remove math and numpy imports from autolev parser output
An example output:

```python
mport sympy.physics.mechanics as me
import sympy as sm
import math as m
import numpy as np

frame_c = me.ReferenceFrame('c')
frame_d = me.ReferenceFrame('d')
frame_f = me.ReferenceFrame('f')
fd, dc = me.dynamicsymbols('fd dc')
fdd, dcd = me.dynamicsymbols('fd dc', 1)
fdd2, dcd2 = me.dynamicsymbols('fd dc', 2)
r, l = sm.symbols('r l', real=True)
point_o = me.Point('o')
point_e = me.Point('e')
frame_d.orient(frame_f, 'Axis', [fd, frame_f.x])
frame_c.orient(frame_d, 'Axis', [dc, frame_d.y])
frame_c.set_ang_vel(frame_f, (frame_c.ang_vel_in(frame_f)).express(frame_f))
point_e.set_pos(point_o, r*frame_d.y-l*frame_c.x)
point_e.set_pos(point_o, (point_e.pos_from(point_o)).express(frame_d))
point_e.set_vel(frame_f, ((point_e.pos_from(point_o)).dt(frame_f)).express(frame_d))
point_e.set_acc(frame_f, ((point_e.vel(frame_f)).dt(frame_f)).express(frame_d))
```

The math input should never be needed and the numpy import is only needed if the parser flag for numerics is set.

### PR DESCRIPTION
**Pull request by sudz123**
Fixes #15186 

#### Release Notes

NO ENTRY

### CODE REVIEW

**Comment by sudz123** [comment]
Oh right, sorry I just realised that. I'll remove this PR and put up a new one

**Comment by NikhilPappu** [comment]
@sudz123 Please don't create duplicate PRs! Make the changes here.

**Comment by sudz123** [comment]
@NikhilPappu so I should remove this line: `self.output_code.append("import math as m\n")` and for the numpy import, where and how should the flag be declared? I am new to this and still trying to get familiar with the code, so thanks for helping out.

**Comment by sudz123** [comment]
I already closed this one, I will be careful in the future, is there no way to re-open this?

**Comment by NikhilPappu** [comment]
@sudz123 You should modify the import lines like you mentioned and also all the places where it is referenced. Just make a boolean value in the Listener class and set it to False by default. Pass this as a parameter to the `parse_autolev` function in `_parse_autolev_antlr.py` and `__init__.py` so users can set it.

**Comment by NikhilPappu** [comment]
@sudz123 Its not quite there. The flag isn't doing anything right now. You need to look at all the strings which have `np.` and `m.`. The parser basically writes these outputs. You should remove anything that uses `.m` as this import is removed. For `np.` you need to use the flag to include it.

**Comment by NikhilPappu** [comment]
@sudz123 Looks like the math import isn't used anywhere so you should just change the `np.`s.

**Comment by NikhilPappu** [comment]
You don't have to set the flag outside the initialiser. Look at the
include_pydy flag. It should work similarly.

File: sympy/parsing/autolev/__init__.py [hunk 1]
```diff
@@ -47,10 +47,9 @@ def parse_autolev(autolev_code, include_numeric=False, numpy_import=False):
     ... "CODE DYNAMICS() some_filename.c")
     >>> my_al_text = '\\n'.join(my_al_text)
     >>> from sympy.parsing.autolev import parse_autolev
-    >>> print(parse_autolev(my_al_text, include_numeric=True))
+    >>> print(parse_autolev(my_al_text, include_numeric=True, numpy_import=True))
```
Comment (moorepants) [inline sympy/parsing/autolev/__init__.py]: I thought the `include_numeric` flag added the numpy related code. I don't think we need an additional flag.

Comment (sudz123) [reply sympy/parsing/autolev/__init__.py]: So currently does the issue still exist? Isn't the numpy related code currently added regardless of any flag?

Comment (moorepants) [reply sympy/parsing/autolev/__init__.py]: Some imports are not needed if include_numeric is false.

Comment (moorepants) [reply sympy/parsing/autolev/__init__.py]: If `include_numeric` is False, there should be no numpy or pydy imports in the generated code. The only imports should be from sympy.

**Comment by sudz123** [comment]
I removed `numpy_import` and set `include_numeric` as a flag for numpy. @NikhilPappu so does this work? Should I proceed to re-write the test examples?

**Comment by sudz123** [comment]
Also while building the test examples, in `test_autolev.py` aren't we passing `include_numeric` as true always. In line : 
`generated_code = parse_autolev(f, include_numeric=True)`

**Comment by moorepants** [comment]
You may need to disable the include_numeric for some tests. Only include if it needs numeric codes.

**Comment by czgdp1807** [comment]
@sudz123 Are you still working on this?

**Comment by kenluck2001** [comment]
Can I work on this PR?

**Comment by czgdp1807** [comment]
@kenluck2001 I think you should proceed with a new PR to complete this one, if the author doesn't respond for 72 hours.

**Comment by kenluck2001** [comment]
@czgdp1807 I have taken over. I will proceed to PR soon.

**Comment by czgdp1807** [comment]
Closing this PR in favour of https://github.com/sympy/sympy/pull/17753
Thank you all for the contributions. Please let me know if it is needed to be opened again for some reasons.

**Pull request closed without merging**